### ISSUE TITLE
EC2 instances with termination protection cannot be destroyed

### ISSUE THREAD

**Issue by ztmr** [pulumi/pulumi-aws#2109]
### What happened?

I am unable to destroy EC2 instance that has termination protection enabled on it.

Mostly the same issue as reported in Terraform AWS provider here: https://github.com/hashicorp/terraform-provider-aws/pull/19277

We have a Git-tracked config file to allow team members to easily add/change/remove instances as a self-service. These instances are all configured with termination protection (so that AWS Console users are not messing with the IaaC stuff). An attempt to remove any of these no longer needed instances, will fail on termination protection being enabled.

The current behavior is good if you have production environment and accidentally attempt to destroy something, so it would be great if we can have some flag to allow unsetting termination protection prior destroying instances -- similar to what suggested in the Terraform issue.

Unlike manual changes in console, I believe that code changes should be tracked and reviewed more carefully, so if the code says to destroy, the resource should be destroyed. Accidental destructive actions should be handled separately using some form of automated review, using tools like Open Policy Agent, or similar.

### Steps to reproduce

* create a new EC2 instance using Pulumi, setting `disableApiTermination`
* attempt to destroy that instance without explicitly unsetting `disableApiTermination`

### Expected Behavior

I'd like to destroy the instance without having to remove the termination protection in the code itself.

Alternatively, I can imagine having a flag that would globally set a default behavior between the current behavior and what is being proposed here, so that anyone can choose what works for them.

I'm also wondering if there is any way of doing pre/post hooks, where we can run a custom code -- for example, unset termination on all the instances to be deleted, and then deleting. Not sure how would that work, but would be cool if possible ;-) Might be related to https://github.com/pulumi/pulumi/issues/1691?

### Actual Behavior

I need to make two code and reconciliation passes:
* remove termination protection flag
* delete the instance itself

### Output of `pulumi about`

```console
$ pulumi about
CLI          
Version      3.38.0
Go Version   go1.19
Go Compiler  gc

Plugins
NAME    VERSION
aws     4.10.0
docker  3.0.0
nodejs  unknown

Host     
OS       ubuntu
Version  20.04
Arch     x86_64

This project is written in nodejs: executable='/usr/bin/node' version='v14.17.2'

Dependencies:
NAME            VERSION
@pulumi/pulumi  3.6.0
@types/js-yaml  4.0.1
@types/node     10.17.60
js-yaml         4.1.0
@pulumi/aws     4.10.0
@pulumi/awsx    0.30.0

$ 
```

### Additional context

Should we consider the same for stop protection? Not sure...

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by stack72** [comment]
Hi @ztmr 

Please can you tell me what version of the provider you are using?

Paul

**Comment by ztmr** [comment]
@stack72 apologies for not sharing that earlier, just updated the issue body now.

I haven't realized that you're actually using the Terraform provider itself... so if my issue is [fixed in TF code >4.16](https://github.com/hashicorp/terraform-provider-aws/pull/19277/commits/1f88e380ecfe5fc888e9afa953e9c1a45068d2ab), it should be also [fixed in >=5.13](https://github.com/pulumi/pulumi-aws/commit/7dbed999d928be64bc28822c4b11422454417c30) of Pulumi, right? I'll try to ugprade to the latest and let you know.

**Comment by stack72** [comment]
Hi @ztmr 

We extract the TF CRUD schema and functionality from the TF provider - we don't actually use the provider after that. We have based Pulumi-AWS v5.13 on the schema / CRUD of TF AWS 4.16 so your issue should be solved with that as you said :) - please do let me know if this isn't then we can reopen the issue

Paul
